designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentService.scala

```diff
@@ -225,6 +235,44 @@ class DeploymentService(
     } yield dmCommand
   }
 
+  private def prepareNodesDeploymentData(
+      command: CommonDeploymentCommand,
+      processDetails: ScenarioWithDetailsEntity[CanonicalProcess],
+      canonicalProcess: CanonicalProcess,
+      actionName: ScenarioActionName
+  )(implicit user: LoggedUser): Future[NodesDeploymentData] = command.nodesDeploymentData match {
+    case Some(data) => Future.successful(data)
+    case None =>
```

Comment:
I thought about always merging defaults into existing parameters (we do this that way for node parameters), but we can try with this solution as well